omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,142 @@
+ConditionController.$inject = ['$scope', 'RestfulService', 'ConditionModel', 'ConceptModel', 'CommonFunctions'];
+
+function ConditionController($scope, RestfulService, ConditionModel, ConceptModel, CommonFunctions) {
+    var self = this;
+
+    $scope.patientUuid = null;
+    $scope.condition = null;
+    const inactiveStatus = 'INACTIVE';
+
+    // this is required inorder to initialize the Restangular service
+    RestfulService.setBaseUrl('/' + OPENMRS_CONTEXT_PATH + '/ws/rest/emrapi');
+
+    /**
+     * Perform POST save
+     * PS: Make sure the 'Non Coded Uuid' setting (under Settings->Condition List) is set inorder to work with noncoded concepts,
+     * the LOCALE needs to be set (most concepts have 'en')
+     * @type {Function}
+     */
+    self.saveCondition = self.saveCondition || function () {
+        if($scope.condition.onSetDate>$scope.condition.endDate){
+                emr.errorAlert("End Date can't be before Onset Date");
+        }
+        else{
+            var conditions = [];
+            conditions.push($scope.condition);
+            RestfulService.post('condition', conditions, function (data) {
+                emr.successAlert("Condition edited successfully")
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            $scope.condition = JSON.parse($scope.condition.split("%22").join("\"").split("%20").join(" "));
+
+            $scope.concept = {
+                display: $scope.condition.concept.name,
+                concept: {
+                    display: $scope.condition.concept.name,
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value = self.convertForDateValue($scope.condition.onSetDate);
+          
+            if($scope.condition.status==inactiveStatus){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = self.convertForDateValue($scope.condition.endDate);
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[0].disabled = true;
+            inputs[1].value = self.convertForDateDisplay($scope.condition.onSetDate);  
```

Comment:
What does convertForDateDisplay mean? Have you tried to look at the naming convention for utility libraries which convert dates to strings?

Reply:
convertForDateDisplay is to display the dates in the onset date and end date fields in the following format- 18 Jun 2019
> Have you tried to look at the naming convention for utility libraries which convert dates to strings?
I didn't understand this. Could you point to the utility library you are  referring to ?

Reply:
Can you google with this text? "javascript library for date formatting"

Reply:
Updated the name.